packages/core/src/routes/one-time-tokens.ts

```diff
@@ -59,4 +68,41 @@ export default function oneTimeTokenRoutes<T extends ManagementApiRouter>(
       return next();
     }
   );
+
+  router.post(
+    '/one-time-tokens/verify',
+    koaGuard({
+      body: OneTimeTokens.guard.pick({
+        token: true,
+        email: true,
+      }),
+      response: OneTimeTokens.guard,
+      status: [200, 400, 404],
+    }),
+    async (ctx, next) => {
+      const { token, email } = ctx.guard.body;
+      const oneTimeToken = await findActiveOneTimeTokenByEmailAndToken(email, token);
+
+      assertThat(
+        oneTimeToken,
+        new RequestError({ code: 'one_time_token.active_token_not_found', status: 404 })
+      );
```

Comment:
Should we fetch the token record by token only first, and then assert if the email in token record equals to the email passed in through params?
```ts
const oneTImeToken = await findActiveOneTimeToken(token);

assertThat(oneTimeToken, new RequestError({ code: 'one_time_token.active_token_not_found', status: 404 });
assertThat(oneTimeToken.email === email, new RequestError({ code: 'one_time_token.email_mismatch' });
```

This way, we can throw more accurate error messages when the emails are mismatch.

What do you think?